bundles/org.eclipse.packagedrone.utils.rpm/src/org/eclipse/packagedrone/utils/rpm/build/PayloadRecorder.java

```diff
@@ -94,12 +118,134 @@ public PayloadRecorder ( final boolean autoFinish ) throws IOException
 
             this.payloadCounter = new CountingOutputStream ( this.fileStream );
 
-            final GZIPOutputStream payloadStream = new GZIPOutputStream ( this.payloadCounter );
+            final OutputStream payloadStream;
+
+            this.payloadCoding = payloadCoding;
+
+            this.payloadFlags = Optional.ofNullable ( payloadFlags );
+
+            switch ( this.payloadCoding )
+            {
+                case "gzip":
+                    GzipParameters parameters = new GzipParameters ();
```

Comment:
Could you make the local variables `final`, unless really necessary. You can add a block `{…}` for the case labels as well in order to scope this.

Reply:
OK